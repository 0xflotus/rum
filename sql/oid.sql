set enable_seqscan=off;

CREATE TABLE test_oid (
	i oid
);

INSERT INTO test_oid VALUES (0),(1),(2),(3),(4),(5);

CREATE INDEX idx_oid ON test_oid USING rum (i);

SELECT * FROM test_oid WHERE i<3::oid ORDER BY i;
SELECT * FROM test_oid WHERE i<=3::oid ORDER BY i;
SELECT * FROM test_oid WHERE i=3::oid ORDER BY i;
SELECT * FROM test_oid WHERE i>=3::oid ORDER BY i;
SELECT * FROM test_oid WHERE i>3::oid ORDER BY i;

EXPLAIN (costs off)
SELECT *, i <=> 0::oid FROM test_oid ORDER BY i <=> 0::oid;
SELECT *, i <=> 0::oid FROM test_oid ORDER BY i <=> 0::oid;

EXPLAIN (costs off)
SELECT *, i <=> 1::oid FROM test_oid WHERE i<1::oid ORDER BY i <=> 1::oid;
SELECT *, i <=> 1::oid FROM test_oid WHERE i<1::oid ORDER BY i <=> 1::oid;
