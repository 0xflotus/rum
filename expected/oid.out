set enable_seqscan=off;
CREATE TABLE test_oid (
	i oid
);
INSERT INTO test_oid VALUES (0),(1),(2),(3),(4),(5);
CREATE INDEX idx_oid ON test_oid USING rum (i);
SELECT * FROM test_oid WHERE i<3::oid ORDER BY i;
 i 
---
 0
 1
 2
(3 rows)

SELECT * FROM test_oid WHERE i<=3::oid ORDER BY i;
 i 
---
 0
 1
 2
 3
(4 rows)

SELECT * FROM test_oid WHERE i=3::oid ORDER BY i;
 i 
---
 3
(1 row)

SELECT * FROM test_oid WHERE i>=3::oid ORDER BY i;
 i 
---
 3
 4
 5
(3 rows)

SELECT * FROM test_oid WHERE i>3::oid ORDER BY i;
 i 
---
 4
 5
(2 rows)

EXPLAIN (costs off)
SELECT *, i <=> 0::oid FROM test_oid ORDER BY i <=> 0::oid;
              QUERY PLAN              
--------------------------------------
 Index Scan using idx_oid on test_oid
   Order By: (i <=> '0'::oid)
(2 rows)

SELECT *, i <=> 0::oid FROM test_oid ORDER BY i <=> 0::oid;
 i | ?column? 
---+----------
 0 |        0
 1 |        1
 2 |        2
 3 |        3
 4 |        4
 5 |        5
(6 rows)

EXPLAIN (costs off)
SELECT *, i <=> 1::oid FROM test_oid WHERE i<1::oid ORDER BY i <=> 1::oid;
              QUERY PLAN              
--------------------------------------
 Index Scan using idx_oid on test_oid
   Index Cond: (i < '1'::oid)
   Order By: (i <=> '1'::oid)
(3 rows)

SELECT *, i <=> 1::oid FROM test_oid WHERE i<1::oid ORDER BY i <=> 1::oid;
 i | ?column? 
---+----------
 0 |        1
(1 row)

